checker/src/main/java/org/checkerframework/checker/nullness/messages.properties

```
@@ -22,6 +22,7 @@ nulltest.redundant=redundant test against null; "%s" is non-null
 instanceof.nullable=instanceof is only true for a non-null expression
 instanceof.nonnull.redundant=redundant @NonNull annotation on instanceof
 new.array.type.invalid=annotations %s may not be applied as component type for array "%s"
+new.array.nullable.ignored=array type annotation must be non-null
 new.class.type.invalid=the annotations %s do not need be applied in object creations
 nullness.on.constructor=do not write nullness annotations on a constructor, whose result is always non-null
```

```suggestion
new.class.type.invalid=the annotations %s do not need be applied in object creations
nullness.on.constructor=do not write nullness annotations on a constructor, whose result is always non-null
nullness.on.new.array=do not write nullness annotations on an array creation, which is always non-null
```

What do you think of using this key? It would follow the existing `nullness.on` errors.
All of these should be split into the invalid use of `@Nullable` and the redundant use of `@NonNull` #966.

An alternative would be to go with the earlier suggestion of `new.array.type.invalid` and split that up/reword to allow errors on the created object or the component type.
In general, it would probably have been nicer to consistently use `xxx.type.invalid` for all of these...

The new `ignored` category I don't like so much - we should just raise an error for the misuse of `@Nullable` and not ignore it.

---

The `nullness.on` convention makes sense. Rewording `new.array.type.invalid` may overload its meaning and. Will be looking at #966 next to split messages up.